File: LeetCode/1768. 交替合并字符串.go
func mergeAlternately(word1 string, word2 string) string {
    w1 :=len(word1)
    w2 := len(word2)
    index1,index2 := 0,0
    sb := strings.Builder{}

    for index1 < w1 || index2 < w2 {
        if index1 < w1 {
        sb.WriteByte(word1[index1])
        index1++
        }
        if index2 < w2 {
        sb.WriteByte(word2[index2])
        index2++
        }
    }
    return sb.String()
}

File: LeetCode/1768. 交替合并字符串.cpp
#include<string>

using namespace std;

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int m = word1.size(), n = word2.size();
        int i = 0, j = 0;
        string answer;
        answer.reserve(m + n);
        while ((i < m) || (j < n))
        {
            if (i < m) {
                answer.push_back(word1[i++]);
            }
            if (j < n) {
                answer.push_back(word2[j++]);
            }
        }
        return answer;
    }
};